source/DataAccess/S3/S3Writer.ts

```diff
@@ -37,36 +37,44 @@ export default class S3Writer implements IDataWriter {
 
     public async ingest(stream: Readable, folderPath: string, fileNamePrefix: string) {
         let dataBody = stream;
+        const dataFile = this.isDataFile(fileNamePrefix);
         let extension = '';
 
-        if (this.shouldEncodeInput(fileNamePrefix)) {
+        if (dataFile) {
             const encodingResult = this._outputEncoder.encode(stream);
             dataBody = encodingResult.outputStream;
             extension = '.' + encodingResult.extension.replace('.', '');
         }
 
-        const parms = {
+        const params = {
             Body: dataBody,
             Bucket:  this._bucketPath + '/' + folderPath,
-            Key: fileNamePrefix + '_' + crypto.randomBytes(8).toString('hex') + extension
+            // No explicit extension if not a data file
+            Key: `${fileNamePrefix}${dataFile ? '_' + crypto.randomBytes(8).toString('hex') + extension : ''}`
```

Comment:
Data files get a randomized tag along with .gz - we don't want this on manifest.json, ddl, or metadata